clerk/keeper.go

```diff
@@ -166,6 +220,13 @@ func GetEventRecordKey(stateID uint64) []byte {
 	return append(StateRecordPrefixKey, stateIDBytes...)
 }
 
+// GetEventRecordKeyWithTime appends prefix to state id and record time
+func GetEventRecordKeyWithTime(stateID uint64, recordTime time.Time) []byte {
+	stateIDBytes := []byte(strconv.FormatUint(stateID, 10))
+	recordTimeBytes := sdk.FormatTimeBytes(recordTime)
+	return append(StateRecordPrefixKeyWithTime, append(recordTimeBytes, stateIDBytes...)...)
```

Comment:
We should create `GetEventRecordTimeKey` like following:

```golang
func EventRecordByTimeKey(t time.Time) []byte {
    return append(StateRecordPrefixKeyWithTime, sdk.FormatTimeBytes(t)...)
}
```

So, it can be used every where like in iterator for `fromTime` and `toTime`